w7/assignment7_4_1/animals.cpp

```diff
@@ -0,0 +1,151 @@
+#include <iostream>
+#include <vector>
+
+// Base class
+class Animal
+{
+public:
+    // Default constructor.
+    Animal(){};
```

Comment:
if you do a definition (ie. the {} after Animal()), you don't need the ; <- these are for declarations, not definitions. I think that it isn't syntactically wrong, but only a bit misgiving! 🙂